app/src/org/commcare/activities/connectId/fragments/PhoneFragment.java

```diff
@@ -0,0 +1,315 @@
+import org.commcare.connect.network.ApiConnectId;
+import org.commcare.connect.network.ConnectNetworkHelper;
+import org.commcare.connect.network.IApiCallback;
+import org.commcare.dalvik.R;
+import org.commcare.utils.KeyboardHelper;
+import org.commcare.utils.PhoneNumberHelper;
+import org.javarosa.core.services.Logger;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Locale;
+
+/**
+ * A simple {@link Fragment} subclass.
+ * Use the {@link PhoneFragment#newInstance} factory method to
+ * create an instance of getContext() fragment.
+ */
+public class PhoneFragment extends Fragment {
+    private TextView titleTextView;
+    private TextView messageTextView;
+    private AutoCompleteTextView countryCodeInput;
+    private AutoCompleteTextView phoneInput;
+    private TextView availabilityTextView;
+    private Button button;
+    private String method;
+    private String existingPhone;
+
+    protected boolean skipPhoneNumberCheck = false;
+
+    TextWatcher watcher = new TextWatcher() {
+        @Override
+        public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+        }
+
+        @Override
+        public void onTextChanged(CharSequence s, int start, int before, int count) {
+            checkPhoneNumber();
+        }
+
+        @Override
+        public void afterTextChanged(Editable s) {
+        }
+    };
+
+    public PhoneFragment() {
+        // Required empty public constructor
+    }
+
+    public static PhoneFragment newInstance() {
+        PhoneFragment fragment = new PhoneFragment();
+        Bundle args = new Bundle();
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for getContext() fragment
+        String existing;
+        View view = inflater.inflate(R.layout.screen_connect_primary_phone, container, false);
+        titleTextView = view.findViewById(R.id.connect_primary_phone_title);
+        messageTextView = view.findViewById(R.id.connect_primary_phone_message);
+        countryCodeInput = view.findViewById(R.id.connect_primary_phone_country_input);
+        phoneInput = view.findViewById(R.id.connect_primary_phone_input);
+        availabilityTextView = view.findViewById(R.id.connect_primary_phone_availability);
+        button = view.findViewById(R.id.connect_primary_phone_button);
+        countryCodeInput.addTextChangedListener(watcher);
+        phoneInput.addTextChangedListener(watcher);
+        button.setOnClickListener(v -> handleButtonPress());
+        requireActivity().setTitle(getString(R.string.connect_phone_page_title));
+//        method = getIntent().getStringExtra(ConnectConstants.METHOD);
```

Comment:
We'll still need these two variables (method and existingPhone) that get passed in by the caller. Perhaps the best way to handle this for now is to pass them in to the newInstance() function?

In the eventual Navigation implementation, this data gets passed in via SafeArgs. See ConnectDownloadingFragment.onCreate for an example.